### PR TITLE
fix(antigravity): show drained models + consolidate quota pools

### PR DESCRIPTION
**Pull request by validatedev**
## Description

This pull request refactors how model quota information is pooled, deduplicated, and displayed in the antigravity plugin. The main changes are the introduction of model "pools" (e.g., "Gemini Pro", "Gemini Flash", "Claude") for grouping and reporting quota status, improved handling of models with missing or depleted quota info, and comprehensive updates to tests and configuration to match the new pooling logic.

**Model pooling and quota reporting:**

* Added a `poolLabel` function to group models into pools ("Gemini Pro", "Gemini Flash", "Claude") for quota tracking and reporting, replacing previous fine-grained model labels.
* Updated deduplication logic to pool models by normalized pool label, keeping the variant with the lowest remaining quota (worst-case), and treating missing quota info as fully depleted (100% used). [[1]](diffhunk://#diff-d9bdb102480d440f0918356be145b4a7929bdd2ceeea09b5bb014110278443c8R291-R301) [[2]](diffhunk://#diff-d9bdb102480d440f0918356be145b4a7929bdd2ceeea09b5bb014110278443c8L355-R370)

**Configuration and display updates:**

* Changed the progress lines in `plugin.json` to use pooled labels ("Gemini Pro", "Gemini Flash", "Claude") instead of specific model versions.

**Test suite overhaul:**

* Updated all tests to use the new pooled labels and logic, including:
  - Pooling and deduplication assertions [[1]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L200-R202) [[2]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L218-R217) [[3]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L236-R231)
  - Handling of missing or empty labels and depleted models [[1]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L309-R372) [[2]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L334-R383) [[3]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L343-R401) [[4]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L362-R418) [[5]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L478-R528) [[6]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L582-R631) [[7]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L614-R663) [[8]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L631-R680) [[9]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L643-R697)
  - Ensuring correct behavior with new test data and edge cases [[1]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L1065-R1118) [[2]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L1107-R1160) [[3]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L1148-R1201) [[4]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L1189-R1242) [[5]](diffhunk://#diff-3572110fe5bc7018242a91a04438ab6c0aa6934a97839fd88c893e5855eac349L1223-R1273)

**Test data adjustments:**

* Updated test model configs and expected results to match new pooling and quota logic, including new model names, quota fractions, and reset times.

These changes ensure that quota tracking is robust, user-facing labels are consistent, and the test suite fully validates the new pooling and depletion logic.

## Type of Change

- [x] Bug fix
- [x] New feature

## Testing

- [x] I ran `bun run build` and it succeeded
- [x] I ran `bun run test` and all tests pass
- [x] I tested the change locally with `bun tauri dev`

## Screenshots

CleanShot 2026-02-21 at 22 28 34

## Checklist

- [x] I read [CONTRIBUTING.md](../CONTRIBUTING.md)
- [x] My PR targets the `main` branch
- [x] I did not introduce new dependencies without justification

---
## Summary by cubic
Consolidated quota reporting into model pools (Gemini Pro, Gemini Flash, Claude) and display models without quota info as fully depleted. This simplifies the UI and keeps quota status accurate across LS and Cloud Code.

- **Refactors**
  - Grouped models into pools via poolLabel and deduplicated by pool using the lowest remaining fraction; all non-Gemini models share the "Claude" pool.
  - Trimmed/normalized labels and skipped configs with missing or empty labels.
  - Updated plugin.json progress lines to pooled labels: Gemini Pro, Gemini Flash, Claude.

- **Bug Fixes**
  - Treated models without quotaInfo as fully depleted (used=100, limit=100), preserving resetTime only when present.
  - Clamped invalid fractions (>1 → 0% used, <0 → 100% used) and ensured ordering.
  - Expanded tests to cover pooling, depleted-variant dedup, and both LS and Cloud Code paths.

<sup>Written for commit f1d8b3a071f33c755a0599d79ce27208350a949a. Summary will update on new commits.</sup>

### CODE REVIEW

File: plugins/antigravity/plugin.js [hunk 1]
```diff
@@ -280,14 +288,17 @@
...
-          remainingFraction: qi.remainingFraction,
-          resetTime: qi.resetTime,
+      var frac = (qi && typeof qi.remainingFraction === "number") ? qi.remainingFraction : 0
+      var rtime = (qi && qi.resetTime) || undefined
+      var pool = poolLabel(normalizeLabel(label))
+      if (!deduped[pool] || frac < deduped[pool].remainingFraction) {
+        deduped[pool] = {
+          label: pool,
+          remainingFraction: frac,
+          resetTime: rtime,
```
[the remark this answers is not in the document]
Comment (validatedev) [reply plugins/antigravity/plugin.js:301]: Not a bug. Within each pool, all models share the same resetTime. No need to overengineer the solution I believe.

**Comment by davidarny** [comment]
Copilot inline comments reviewed by CPT-5.3-Codex.

None are actionable for this PR:
- [r2836542197](https://github.com/robinebers/openusage/pull/204#discussion_r2836542197) conflicts with the intentional 3-pool design (Gemini Pro / Gemini Flash / Claude).
- [r2836542210](https://github.com/robinebers/openusage/pull/204#discussion_r2836542210) conflicts with the intentional worst-case pooled quota semantics validated by tests.
- [r2836542223](https://github.com/robinebers/openusage/pull/204#discussion_r2836542223) is a duplicate wording nit; the existing code comment already states the implemented behavior clearly.

File: plugins/antigravity/plugin.js [hunk 2]
```diff
@@ -251,6 +251,14 @@
     return label.replace(/\s*\([^)]*\)\s*$/, "").trim()
   }
 
+  function poolLabel(normalizedLabel) {
+    var lower = normalizedLabel.toLowerCase()
+    if (lower.indexOf("gemini") !== -1 && lower.indexOf("pro") !== -1) return "Gemini Pro"
+    if (lower.indexOf("gemini") !== -1 && lower.indexOf("flash") !== -1) return "Gemini Flash"
+    // Claude, GPT-OSS, and any other non-Gemini models share the "Claude" pool
+    return "Claude"
```
[the remark this answers is not in the document]
Comment (validatedev) [reply plugins/antigravity/plugin.js]: This is intentional. All non-Gemini models share a single quota pool. Their remainingFraction and resetTime are always identical because they draw from the same bucket.

**Comment by davidarny** [comment]
> Copilot inline comments reviewed by CPT-5.3-Codex.
> 
> None are actionable for this PR:
> 
> * [r2836542197](https://github.com/robinebers/openusage/pull/204#discussion_r2836542197) conflicts with the intentional 3-pool design (Gemini Pro / Gemini Flash / Claude).
> * [r2836542210](https://github.com/robinebers/openusage/pull/204#discussion_r2836542210) conflicts with the intentional worst-case pooled quota semantics validated by tests.
> * [r2836542223](https://github.com/robinebers/openusage/pull/204#discussion_r2836542223) is a duplicate wording nit; the existing code comment already states the implemented behavior clearly.

@validatedev so basically copilot's comments seems to be false positives. Approving!

**Review by davidarny — DISMISSED** [review]
LGTM!

**Comment by validatedev** [comment]
@davidarny thanks! yes they’re mostly false-positives

i named the vertex quota as “claude” as who uses gpt-oss lol

if you have better ideas i'd like to hear them!

**Comment by davidarny** [comment]
> as who uses gpt-oss lol

😂 true

**Review by robinebers — APPROVED** [review]
Great change, LGTM
